### ISSUE TITLE
Fleet Agent Fails to Install with Fleet Server on macOS

### ISSUE THREAD

**Issue by charlie-pichette** [elastic/beats#25563]
Please post all questions and issues on https://discuss.elastic.co/c/beats
before opening a Github Issue. Your questions will reach a wider audience there,
and if we confirm that there is a bug, then you can open a new issue.

For security vulnerabilities please only send reports to security@elastic.co.
See https://www.elastic.co/community/security for more information.

Please include configurations and logs if available.

For confirmed bugs, please report:
- Version: 7.13.0 BC4 build 5c4bc719
- Operating System: macOS High Sierra, Mojave, Catalina and Big Sur
- Discuss Forum URL:
- Steps to Reproduce: 
  - Setup a new 7.13.0 BC4 build 5c4bc719 environment in cloud staging
  - Basically follow these steps:
    - Staging Cloud UI Steps
      1. Setup a new deployment on the staging cloud with 7.13.0
      1. Save the credentials it gives you (elastic:<password>)
      1. After the deployment is done being created, click on "Copy Cluster ID" next to Elasticsearch, save it as the ES cluster id in notes somewhere.
      1. Using the cluster id, construct a complete URL for ES like so: https://<copied ES cluster id>.us-central1.gcp.foundit.no:9243   --- save this in your notes as well, and change everything that follows <copied ES cluster id> accordingly if you are using a different cloud provider / location for the deployment
      1. Now we can progress to configuring the actual Kibana Fleet UI, click on "Launch" next to Kibana to open Kibana and login accordingly if necessary.
    - Kibana UI Steps
      1. Click the 3 stacked lines in the top left to open the Kibana Menu
      1. Scroll down and click on "Fleet" under the "Management" section
      1. Click on "Policies" tab
      1. Click "Create agent policy"
      1. Name it accordingly, ex. "endpoint-test-policy"
      1. Click "Create agent policy"
      1. Click on the new policy you just created in the list
      1. Click "Actions" > "View policy"
      1. Copy the topmost value for "id" and save in notes as "policy id"
      1. Click "close" in the bottom left of the sub-page
      1. Click "Add integration"
      1. Search for "Endpoint" and select "Endpoint Security"
      1. Name the integration something useful like "test-endpoint-1" or "endpoint-1"
      1. Click "Save integration"
      1. Once it is done saving, click "Add integration" again
      1. Search for "Fleet" and select "Fleet Server"
      1. Change the name of the integration if desired
      1. Expand the options next to the checked "Fleet Server" row
      1. Under the "Host" section, change it to "https://localhost"
      1. Click "Save integration"
      1. Now click the "Agents" tab
      1. Click "Add agent"
      1. Make sure it is on the "Enroll in Fleet" tab
      1. Click "Generate Service Token"
      1. Save the generated service token in your notes
    - Test Endpoint Steps
      1. On the endpoint machine where you wish to install agent and test, download the appropriate agent package from the BC build page (https://staging.elastic.co/7.13.0-24571870/summary-7.13.0.html)
      1. Unpack the agent package on the machine
      1. Navigate to the unpacked directory and run the following command
```
sudo ./elastic-agent install --fleet-server-es=<saved ES URL from notes> --fleet-server-policy=<saved policy id from notes> --fleet-server-service-token=<saved service token from notes>
```

The result is:
```
The Elastic Agent is currently in BETA and should not be used in production

2021-05-05T13:51:57.602-0400	INFO	cmd/enroll_cmd.go:300	Generating self-signed certificate for Fleet Server
2021-05-05T13:51:59.281-0400	INFO	cmd/enroll_cmd.go:588	Waiting for Elastic Agent to start Fleet Server
2021-05-05T13:52:01.303-0400	INFO	cmd/enroll_cmd.go:621	Fleet Server - Starting
2021-05-05T13:52:03.316-0400	INFO	cmd/enroll_cmd.go:602	Fleet Server - Running on policy with Fleet Server integration: eed4b010-ada4-11eb-8fca-13724031a36c; missing config fleet.agent.id (expected during bootstrap process)
2021-05-05T13:52:03.681-0400	WARN	[transport]	transport/tcp.go:52	DNS lookup failure "mac-highsierra-10-0-6-95": lookup mac-highsierra-10-0-6-95: no such host
Error: fail to enroll: fail to execute request to fleet-server: lookup mac-highsierra-10-0-6-95: no such host
Error: enroll command failed with exit code: 1
```

The problem is on macOS you cannot ping the hostname.  You can ping localhost.  You can ping <hostname>.local but you cannot ping just <hostname>.
```
mac-highsierra-10-0-6-95:~ zeus$ hostname
mac-highsierra-10-0-6-95
mac-highsierra-10-0-6-95:~ zeus$ ping -c 1 localhost
PING localhost (127.0.0.1): 56 data bytes
64 bytes from 127.0.0.1: icmp_seq=0 ttl=64 time=0.050 ms

--- localhost ping statistics ---
1 packets transmitted, 1 packets received, 0.0% packet loss
round-trip min/avg/max/stddev = 0.050/0.050/0.050/0.000 ms
mac-highsierra-10-0-6-95:~ zeus$ ping -c 1 mac-highsierra-10-0-6-95
ping: cannot resolve mac-highsierra-10-0-6-95: Unknown host
mac-highsierra-10-0-6-95:~ zeus$ ping -c 1 mac-highsierra-10-0-6-95.local
PING mac-highsierra-10-0-6-95.local (10.0.6.95): 56 data bytes
64 bytes from 10.0.6.95: icmp_seq=0 ttl=64 time=0.059 ms

--- mac-highsierra-10-0-6-95.local ping statistics ---
1 packets transmitted, 1 packets received, 0.0% packet loss
round-trip min/avg/max/stddev = 0.059/0.059/0.059/0.000 ms
```
The workaround is to edit `/etc/hosts` and add the entry `127.0.0.1  mac-highsierra-10-0-6-95`.

**Comment by ph** [comment]
@charlie-pichette This look like a configuration issue on the mac?

```
Error: fail to enroll: fail to execute request to fleet-server: lookup mac-highsierra-10-0-6-95: no such host
```

Without the change in the `/etc/host` file, I am certain the the **ping** command will fail too

Who is giving you the `mac-highsierra-10-0-6-95` URL, it is fleet?

**Comment by nchaulet** [comment]
Looks similar to the issue @kpollich had yesterday on a newly setup mac, I think he had to enable file sharing to get this working.

**Comment by charlie-pichette** [comment]
@ph The `mac-highsierra-10-0-6-95` is the machines hostname.  Just typing `hostname` at the shell prompt returns that info.

@nchaulet enabling file sharing did not help.

**Comment by charlie-pichette** [comment]
The issue is that I had Fleet integration in the policy.  If I remove that agent installs.  This appears to be an issue between the keyboard and the chair.

**Comment by charlie-pichette** [comment]
I think we can close this issue.

**Comment by ph** [comment]
@charlie-pichette I am happy to close it. you had a fleet integration in a non fleet server policy?

**Comment by charlie-pichette** [comment]
@phg Basically.  @EricDavisX corrected my misunderstanding of how to configure the environment.

**Comment by kpollich** [comment]
> Looks similar to the issue @kpollich had yesterday on a newly setup mac, I think he had to enable file sharing to get this working.

Closing the loop here - this is the same issue I dealt with yesterday on my new work MacBook, yep. DNS resolution was mysteriously just broken out of the box. e.g. `ping $(hostname)` in terminal would not resolve to my local machine.

https://elastic.slack.com/archives/C01EECES05R/p1620148047297300

Eventually found this StackExchange post that solved for me (via macOS sharing preferences), but the `/etc/hosts` workaround is also mentioned as a common fix: https://apple.stackexchange.com/questions/253817/cannot-ping-my-local-machine/253834#253834
